src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Log4Net/AppenderAttachedImplIntegration.cs

```diff
@@ -0,0 +1,66 @@
+// <copyright file="AppenderAttachedImplIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.ComponentModel;
+using Datadog.Trace.ClrProfiler.CallTarget;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Log4Net
+{
+    /// <summary>
+    /// LoggerFactoryScopeProvider.ForEach&lt;TState&gt; calltarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = "log4net",
+        TypeName = "log4net.Util.AppenderAttachedImpl",
+        MethodName = "AppendLoopOnAppenders",
+        ReturnTypeName = ClrNames.Int32,
+        ParameterTypeNames = new[] { "log4net.Core.LoggingEvent" },
+        MinimumVersion = "2.0.0",
+        MaximumVersion = "2.*.*",
+        IntegrationName = "Log4Net")]
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class AppenderAttachedImplIntegration
+    {
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TLoggingEvent">The type of the logging event</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="loggingEvent">The logging event</param>
+        /// <returns>Calltarget state value</returns>
+        public static CallTargetState OnMethodBegin<TTarget, TLoggingEvent>(TTarget instance, TLoggingEvent loggingEvent)
+            where TLoggingEvent : ILoggingEvent
+        {
+            if (!loggingEvent.Properties.Contains(CorrelationIdentifier.TraceIdKey))
+            {
+                loggingEvent.Properties[CorrelationIdentifier.ServiceKey] = CorrelationIdentifier.Service;
+                loggingEvent.Properties[CorrelationIdentifier.VersionKey] = CorrelationIdentifier.Version;
+                loggingEvent.Properties[CorrelationIdentifier.EnvKey] = CorrelationIdentifier.Env;
+                loggingEvent.Properties[CorrelationIdentifier.TraceIdKey] = CorrelationIdentifier.TraceId;
+                loggingEvent.Properties[CorrelationIdentifier.SpanIdKey] = CorrelationIdentifier.SpanId;
```

Comment:
Should we omit these if there's no active trace (instead of adding as `0`)? Omitting them is the approach I took in the `ILogger` integration. 

I think this (should) also need updates to the integration tests right, as it's a change in behaviour from the previous implementation?

Reply:
In the latest commits, I followed your suggestion to omit the span_id / trace_id values from the dictionary when there's no active span